src/Sulu/Bundle/AdminBundle/Resources/js/components/Icon/Icon.js

```diff
@@ -69,6 +82,7 @@ export default class Icon extends React.PureComponent<Props> {
         const onClickProperties = onClick
             ? {
                 onClick: this.handleClick,
+                onKeyPress: this.handleKeypress,
                 role: 'button',
```

Comment:
The change make sense: I think we should also don't set a `role` when there is nothing to click?

```suggestion
                role: onClick ? 'button' : null,
```

If a better solution if its possible is that we do:

```
if (onClick) {
    return <button type="button" {...attributes}></button>
}

return <span {...attributes}></span>;
```

We just need to make sure the css styling works for span and button correcly like before.

Reply:
Maybe then we also don't need the `handleKeypress` as it is then supported natively?

Reply:
This component currently tries to be both a button and a non-interactive icon. The icon therefore also needs hover and focus styles depending on its use. In the long term, the cleanest solution would be to use icons only for decorative purposes and always use them inside native buttons. I can take a look at how often the icon component is used with an onclick prop.

The role does not need to be provided with a ternary operator at this point, since the entire block is already just set when an onClick function is defined (two lines above).

To keep the style changes within this pull request small, I would suggest not changing the element type.

Reply:
I agree here that the `icon` should not have a `onClick` and just should have been wrapped with button where required, but at current state we can not change that as that would be a backward compatibility break. But we could deprecate the `onClick` like [here](https://github.com/sulu/sulu/blob/04e987b42a80ede16c6279cc153df3b409622d6e/src/Sulu/Bundle/AdminBundle/Resources/js/views/List/toolbarActions/UploadToolbarAction.js#L36-L40)
and don't use <Icon onClick in our core components and go there with wrapping them in a button where they are clickable. But also I'm not aware where this is currently and we probably need to check all <Icon usages.

Reply:
I added a deprecation warning and found the `onClick` prop in the following components:
`Block`, `Card`, `Chip`, `ColumnList/ItemButton`, `Input`, `Navigation/Item`, `Input`, `Overlay`, `Snackbar`, `Table`, `Table/Row` and `containers/List`. In all these cases, the icon component should now be keyboard operable.

In another pull request, I can look at restructuring the core components.

Reply:
We should at current state remove then the `deprecation` of onClick. And add it after we did change it in all other cases. So we are sure not the sulu core itself is throwing a deprecation message and that it also work like we expected to remove the onClick here.